### ISSUE TITLE
Turn on XrdStatistics in Configs by Default

### ISSUE THREAD

**Issue by klannon** [matz-e/lobster#386]
I'm not really entirely sure yet how to do this, but thanks to @bbockelm I now at least know that there's a service that monitors XRootD activity and (I think) dumps it into report.xml.  See here:

https://github.com/cms-sw/cmssw/blob/CMSSW_8_0_X/Utilities/XrdAdaptor/src/XrdStatistics.h

**Comment by klannon** [comment]
This looks promising: http://cmslxr.fnal.gov/lxr/source/Utilities/XrdAdaptor/src/XrdStatistics.cc?v=CMSSW_8_0_3#0022

**Comment by matz-e** [comment]
I haven't looked into it yet — but this sounds just like the thing for us!

**Comment by khurtado** [comment]
Nice! I activated it for the `slim.py` example and it produces the following in the FJR:

<PerformanceReport>
  <PerformanceModule Metric="T2_US_Caltech"  Module="XrdSiteStatistics" >
    <Metric Name="read-numOperations" Value="20"/>
    <Metric Name="read-totalMegabytes" Value="0.981539"/>
    <Metric Name="read-totalMsecs" Value="2266.39"/>
    <Metric Name="readv-numChunks" Value="84"/>
    <Metric Name="readv-numOperations" Value="14"/>
    <Metric Name="readv-totalMegabytes" Value="7.40662"/>
    <Metric Name="readv-totalMsecs" Value="6453.42"/>
  </PerformanceModule>

**Comment by bbockelm** [comment]
Yup, that sounds about right!  Are any important statistics or information missing?

**Comment by klannon** [comment]
@bbockelm:  I was hoping to get a list of files opened and from which site.  This doesn't seem to be there.  Are we missing an option?  What about a summary of XRootD errors?

**Comment by annawoodard** [comment]
It [looks ](http://cmslxr.fnal.gov/lxr/source/Utilities/XrdAdaptor/src/XrdStatistics.cc?v=CMSSW_8_0_3#0135)like what @khurtado showed is all that's currently available.

**Comment by bbockelm** [comment]
So, that's a bit tricky.  CMSSW will continuously search for new file sources and start reading from them actively if they prove to perform better than the current set of sources.  So, each file opened will come from multiple sites.

The list of input files is available in a different part of the framework job report.

We don't keep a summary of Xrootd errors, unfortunately: if we recover from an error, CMSSW recovers and moves on.  That would be an excellent (and relatively simple) improvement one could submit back to CMSSW!

**Comment by klannon** [comment]
Honestly, even just having a summary of all the sites read from, even if it doesn't say which files or how much data, would be helpful in trying to get a picture of whether a particular site is causing problems.  So, is that tricky = impossible or tricky = needs some work?

Where would the XRootD error tracking code go?

**Comment by bbockelm** [comment]
A per-site summary is what we have today.

To keep a count of errors that occurred, you would start adding new members to the statistics class - such as here:

https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/Utilities/XrdAdaptor/src/XrdStatistics.cc#L92

then instrument the various places failures could occur.  I think the primary place to hit is this one:

https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/Utilities/XrdAdaptor/src/XrdRequestManager.cc#L783
